spec.html

```diff
@@ -6548,38 +6542,45 @@ <h1>GetBindingValue ( _N_, _S_ )</h1>
 
         <emu-clause id="sec-object-environment-records-deletebinding-n">
           <h1>DeleteBinding ( _N_ )</h1>
-          <p>The concrete Environment Record method DeleteBinding for object Environment Records can only delete bindings that correspond to properties of the environment object whose [[Configurable]] attribute have the value *true*.</p>
+          <p>The DeleteBinding concrete method of an object Environment Record _envRec_ takes argument _N_ (a String). It can only delete bindings that correspond to properties of the environment object whose [[Configurable]] attribute have the value *true*. It performs the following steps when called:</p>
           <emu-alg>
-            1. Let _envRec_ be the object Environment Record for which the method was invoked.
             1. Let _bindings_ be the binding object for _envRec_.
             1. Return ? _bindings_.[[Delete]](_N_).
           </emu-alg>
         </emu-clause>
 
         <emu-clause id="sec-object-environment-records-hasthisbinding">
           <h1>HasThisBinding ( )</h1>
-          <p>Regular object Environment Records do not provide a `this` binding.</p>
+          <p>The HasThisBinding concrete method of an object Environment Record _envRec_ takes no arguments. It performs the following steps when called:</p>
           <emu-alg>
             1. Return *false*.
           </emu-alg>
+          <emu-note>
+            <p>Object Environment Records do not provide a `this` binding.</p>
+          </emu-note>
         </emu-clause>
 
         <emu-clause id="sec-object-environment-records-hassuperbinding">
           <h1>HasSuperBinding ( )</h1>
-          <p>Regular object Environment Records do not provide a `super` binding.</p>
+          <p>The HasSuperBinding concrete method of an object Environment Record _envRec_ takes no arguments. It performs the following steps when called:</p>
           <emu-alg>
             1. Return *false*.
           </emu-alg>
+          <emu-note>
+            <p>Object Environment Records do not provide a `super` binding.</p>
+          </emu-note>
         </emu-clause>
 
         <emu-clause id="sec-object-environment-records-withbaseobject">
           <h1>WithBaseObject ( )</h1>
-          <p>Object Environment Records return *undefined* as their WithBaseObject unless their _withEnvironment_ flag is *true*.</p>
+          <p>The WithBaseObject concrete method of an object Environment Record _envRec_ takes no arguments. It performs the following steps when called:</p>
           <emu-alg>
-            1. Let _envRec_ be the object Environment Record for which the method was invoked.
             1. If the _withEnvironment_ flag of _envRec_ is *true*, return the binding object for _envRec_.
             1. Otherwise, return *undefined*.
           </emu-alg>
+          <emu-note>
+            <p>Object Environment Records return *undefined* as their WithBaseObject unless their _withEnvironment_ flag is *true*.</p>
```

Comment:
I'd rather not include this. It doesn't add anything that can't be easily obtained from reading the algorithm, and I don't like how it's worded.

Reply:
Okay, removing it.